lib/galaxy/model/dataset_collections/matching.py

```
@@ -60,7 +60,10 @@ def __attempt_add_to_linked_match(self, input_name, hdca, collection_type_descri
             self.subcollection_types[input_name] = subcollection_type
 
     def slice_collections(self):
-        return self.linked_structure.walk_collections(self.collections)
+        if self.linked_structure:
+            return self.linked_structure.walk_collections(self.collections)
+        else:
+            return []
```

I think raising an `Exception` would be better instead of silently returning an empty set, yes?

---

Changed back to `assert`. I think it could break some tests, but let's see